### ISSUE TITLE
Add option to set locale on UnitRegistry; auto-Babelise (resp. Babelize!)

### ISSUE THREAD

**Issue by OJFord** [hgrecco/pint#904]
It would be good to be able to set, for example, `locale='en_GB'` on instantiating a `UnitRegistry`, and then have it automatically `format_babel` to that locale.

As it is, the following is a little odd:

```
>>> ureg.litre
<Unit('liter')>
>>> str(3 * ureg.litre)
'3 liter'
```

it's good that both spellings are accepted in Python, but it's a bit odd to then have it come back out in AmE having explicitly used BrE.

Having `ureg = UnitRegistry(locale='en_GB')` could mitigate that, and then also allow `'litre'` in the response of `ureg.cc.compatible_units()`.

I could help with this, if it is something you agree with. Cheers.

**Comment by hgrecco** [comment]
I like this. My suggestions (which are of course open for discussion):
- add a method (`set_locale`) to the following classes: Unit/System/Quantity/BaseRegistry
  This method should take:
  - None to have no locale  (i.e. no transformation).
  - a string with the locale name.
  - 'auto' to detect the current locale
- `set_locale` in BaseRegistry should call the ones Unit/System/Quantity
- add the `locale` argument to `BaseRegistry`, and call the `set_locale` from there.

Alternatively:
- Create a new class in `registry.py` named LocalizedRegistry with the `locale` modeled like the other registry classes that extend functionality.

Can you submit a PR?

**Comment by hgrecco** [comment]
In addition, a better error message to be provided when babel is not installed would be nice.

**Comment by hgrecco** [comment]
Closed by #944
